code/script.js

```diff
@@ -1,325 +1,427 @@
+    )
+  } else {
+    alert(
+      `Nope, the person doesn't have ${value} hair. Remove all that have ${value} hair.`
+    )
+  }
+} else if (category === 'eye') {
+  if (keep) {
+    alert(
+      `Good choice, the person has ${value} eyes! Keep all that has ${value} eyes.`
+    )
+  } else {
+    alert(
+      `Sorry, the person doesn't have ${value} eyes. Remove all that has ${value} eyes.`
+    )
+  }
+} else if (category === 'other') {
+  if (keep) {
+    alert(
+      `You are right, the person is a ${value}! Keep all that are ${value}.`
+    )
+  } else {
+    alert(
+      `Sorry, the person is not a ${value}. Remove all that are ${value}.`
+    )
+  }
+} else {
+  if (keep) {
+    alert(
+      `Yes, the person has ${value}! Keep all that still can be the secret person.`
+    )
+  } else {
+    alert(
+      `No, the person doesnt have ${value}! Remove all people that in fact are not the secret person.`
+    )
+  }
+}
+
+// Determine what is the category are, filter to keep and remove based on the variables from keep
+
+
+// for hair and eyes
+if (category === 'hair' || category === 'eyes') {
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category] === value
+    )
+  } else {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category] !== value
+    )
+  }
+  // for accessories and other
+} else if (category === 'superpower' || category === 'other') {
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) =>
+      person[category].includes(value)
+    )
+  } else {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => !person[category].includes(value)
+    )
+  }
+}
+
+// Here we reinvoke the board with the remaining people after filtering.
+generateBoard()
+}
+
+
+// when clicking guess, the player first have to confirm that they want to make a guess.'
+ // store the interaction from the player in a variable.
+ 
+const guess = (personToConfirm) => {
+  const guessed = confirm(`Do you want to guess who the person is?`)
+  if (guess) {
+    checkMyGuess(personToConfirm)
+  } 
+}
+// If you confirm, this function is invoked if its right guess
+const checkMyGuess = (personToCheck) => {
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `Holycrap you are a mind reader! It was ${secret.name}!`
+    // The sound if they are right
+    var audio = new Audio('wand.mp3');
+    audio.play();
+
+// If the person guessed wrong
+  } else {
+    winOrLoseText.innerHTML = `Im sorry you guessed at the wrong person. The secret person was ${secret.name}. I hope you do better next time you play!`
+
+ // The sound if the are wrong :) 
+    var audio = new Audio('evilLaugh.mp3');
+    audio.play();
+
+  }
+
+  winOrLose.style.display = 'flex'
+
+}
+    
+ // Here we give the player one option i the wanna restart and play again
+const  oneMoreTime = () => {
+  winOrLose.style.display = 'none'
+  start();
+}

```

Comment:
Did you think of going back to the start instead of using this new part oneMoreTime, when clicking PlayAgainButton? You could have used a few less lines if doing so. Just a tip!